lpc55_sign/src/bin/lpc55_sign.rs

```diff
@@ -545,3 +517,69 @@ fn check_plain_image(_image: &[u8]) -> Result<()> {
     check!(OK, "Nothing to check for plain image");
     Ok(())
 }
+
+fn read_certs(paths: &[PathBuf]) -> Result<Vec<Vec<u8>>> {
+    Ok(paths
+        .iter()
+        .map(std::fs::read)
+        .collect::<Result<Vec<Vec<u8>>, _>>()?)
+}
+
+fn write_to_file(path: &PathBuf, bytes: &[u8]) -> Result<()> {
+    Ok(OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(&path)?
+        .write_all(&bytes)?)
+}
+
+fn write_signed_image(
+    image: &ImageArgs,
+    dice: &DiceArgs,
+    signing_certs: &[&[u8]],
+    root_certs: &[&[u8]],
+    private_key: &PathBuf,
+) -> Result<()> {
+    let bin = std::fs::read(&image.src_bin)?;
+    let (stamped, rkth) =
+        signed_image::stamp_image(&bin, signing_certs, root_certs.try_into()?, image.address)?;
+
+    let private_key = std::fs::read_to_string(private_key)?;
+    let signed = signed_image::sign_image(&stamped, &private_key)?;
+
+    let cmpa = signed_image::create_cmpa(
+        dice.with_dice,
```

Comment:
We may just want to pass `dice` into this function, since we're literally unpacking every value from the struct.

Reply:
Indeed, in a8b5677 `generate_cmpa` takes a `DiceArgs`.